compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -333,6 +333,15 @@
     private static final ParserRuleContext[] NEW_KEYWORD_RHS =
             { ParserRuleContext.TYPE_DESC_IN_NEW_EXPR, ParserRuleContext.EXPRESSION_RHS };
 
+    private static final ParserRuleContext[] TABLE_CONSTRUCTOR_OR_QUERY_START =
+            { ParserRuleContext.TABLE_KEYWORD, ParserRuleContext.STREAM_KEYWORD, ParserRuleContext.QUERY_EXPRESSION };
+
+    private static final ParserRuleContext[] TABLE_CONSTRUCTOR_OR_QUERY_RHS =
+            { ParserRuleContext.TABLE_CONSTRUCTOR, ParserRuleContext.QUERY_EXPRESSION };
+
+    private static final ParserRuleContext[] QUERY_EXPRESSION_RHS =
+            { ParserRuleContext.SELECT_CLAUSE, ParserRuleContext.WHERE_CLAUSE, ParserRuleContext.FROM_CLAUSE,
+                    ParserRuleContext.LET_CLAUSE };
```

Comment:
Do we need to add EXPRESSION_RHS as an option here?

Reply:
Do we? Then it will end up in an infinite loop right? QUERY_EXPRESSION_RHS is called as a next rule inside seekMatchInExpressionRhs and then QUERY_EXPRESSION_RHS again call seekMatchInExpressionRhs and go on?

Reply:
I see. How does it match something like a semicolon at the end of the expr?
e.g.:  `... where a > b ;`

Reply:
May be we could add something like END_OF_QUERY, then end the query-expr context inside it, and then go ahead for EXPRESSION_RHS?

btw, I am merging the PR anyway, since the rest of the changes looks fine.